lib/corm/model.rb

```diff
@@ -139,20 +143,74 @@ def self.count
-        statement = "SELECT * FROM #{keyspace}.#{table} WHERE #{fields} LIMIT 1;"
-        statements['get'] = session.prepare statement
+    ##
+    # Find by keys.
+    # This `find` methods wants to be as flexible as possible.
+    #
+    # Unless a block is given, it returns an `Enumerator`, otherwise it yields
+    # to the block an instance of the found Cassandra entries.
+    #
+    # If no keys is passed as parameter, the methods returns (an Enumerator for)
+    # all the results in the table.
+    #
+    # The options hash support the ':limit' option to append at the statement;
+    # the default is no limit.
+    #
+    # The 'key_values' parameter is an Hash where the keys are the "column
+    # names" and the values... are the values.
+    #
+    # If the keys passed as parameter are more than the defined by the table the
+    # query is not valid, it cannot be executed and an error is raised.
+    # Other exceptions are raised when the keys doesn't include all the
+    # (required) partition_keys or the clustering keys doesn't are in the
+    # defined order.
+    def self.find(key_values = {}, query_options = {}, &block)
+
+      raise ArgumentError unless key_values.is_a?(Hash)
+      raise ArgumentError unless query_options.is_a?(Hash)
+
+      unless key_values.empty?
+        raise TooManyKeysError if there_are_too_many_keys_requested?(key_values)
+        raise MissingPartitionKey if a_partition_key_is_missing?(key_values)
+        raise UnknownClusteringKey if an_unknown_clustering_key_is_requested?(key_values)
+        # raise UnknownKey if an_unknown_key_is_requested?(key_values)
+        raise MissingClusteringKey if a_clustering_key_is_missing?(key_values)
       end
-      values = primary_key.flatten.map do |key|
-        relations[key.to_s] || relations[key.to_sym]
+
+      return to_enum(:find, key_values, query_options) unless block_given?
+
+      statement_find_key = Array(query_options.fetch(:statement_key, 'find')).flatten
+      field_names = []
+
+      key_values.each do |key, value|
+
+        statement_find_key << key.to_s
+        field_names << "#{key} = ?"
+      end
+
+      statement_find_key = statement_find_key.join('_')
+      statement_find_key.concat("_limit#{query_options[:limit]}") if query_options[:limit]
+
+      if statements[statement_find_key].nil?
+        statement = self.the_select_statement_for(key_values, field_names, query_options[:limit])
+        statements[statement_find_key] = session.prepare(statement)
+      end
+
+      execute(statements[statement_find_key], arguments: key_values).each do |cassandra_record_|
+        block.call(new(_cassandra_record: cassandra_record_))
       end
-      cassandra_record_ = execute(statements['get'], arguments: values).first
-      cassandra_record_ ? new(_cassandra_record: cassandra_record_) : nil
+    end
+
+    def self.get(relations)
+      query_options = {
+        limit: 1,
+        statement_key: 'get'
```

Comment:
I think that it's in big part some kind of fancy readability... that kind of things that could became a bug...
:smile: